paasta_tools/cli/cmds/status.py

```diff
@@ -1376,53 +1431,139 @@ def create_replica_table(
-                            )
+            if state == ReplicaState.WARMING_UP:
+                if verbose > 0:
+                    warmup_duration = datetime.now().timestamp() - pod.create_timestamp
+                    humanized_duration = humanize.naturaldelta(
+                        timedelta(seconds=warmup_duration)
+                    )
+                    grace_period_remaining = (
+                        pod.create_timestamp
+                        + main_container.healthcheck_grace_period
+                        - datetime.now().timestamp()
+                    )
+                    humanized_remaining = humanize.naturaldelta(
+                        timedelta(seconds=grace_period_remaining)
+                    )
+                    table.append(
+                        PaastaColors.cyan(
+                            f"  Still warming up, {humanized_duration} elapsed, {humanized_remaining} before healthchecking starts"
                         )
-            if state == ReplicaState.HEALTHCHECK_FAILING:
-                # TODO: Use actual container.liveness_probe port or healthcheck from config
+                    )
+        if recent_liveness_failure(pod) and state != ReplicaState.TERMINATING:
+            healthcheck_string = (
+                "check your healthcheck configuration in yelpsoa_configs"
+            )
+            if main_container and main_container.healthcheck_cmd:
+                if main_container.healthcheck_cmd.http_url:
+                    healthcheck_string = (
+                        f"run `curl {main_container.healthcheck_cmd.http_url}`"
+                    )
+                elif main_container.healthcheck_cmd.tcp_port:
+                    healthcheck_string = f"verify your service is listening on {main_container.healthcheck_cmd.tcp_port}"
+                elif main_container.healthcheck_cmd.cmd:
+                    healthcheck_string = f"check why the following may be failing: `{main_container.healthcheck_cmd.cmd}`"
+            table.append(
+                PaastaColors.red(
+                    f"  Healthchecks are failing. To investigate further, {healthcheck_string}"
+                )
+            )
+        if state.is_unhealthy():
+            if verbose < 2:
                 table.append(
                     PaastaColors.red(
-                        f"  To test healthcheck directly, run `curl {pod.ip}:8888/status`"
+                        f"  Consider checking logs with `paasta logs -s {service} -i {instance} -p {pod.name}`"
                     )
                 )
             else:
-                # TODO: Specify full paasta logs cmd once pod filtering is available
-                table.append(
-                    PaastaColors.red(f"  Consider checking logs with `paasta logs`")
-                )
+                if pod.events:
+                    table.extend(format_pod_event_messages(pod.events, pod.name))
```

Comment:
how much does this end up printing?  i could see this being a lot..but maybe we just ship it and try it out internally